src/Psalm/Internal/PhpVisitor/Reflector/ClassLikeNodeScanner.php

```diff
@@ -1161,17 +1163,31 @@ private function visitClassConstDeclaration(
                 $const
             );
 
+            if ($const_type
+                && $const->value instanceof \PhpParser\Node\Expr\BinaryOp\Concat
+                && $const_type->isSingle()
+                && get_class(array_values($const_type->getAtomicTypes())[0]) === Type\Atomic\TString::class
+            ) {
+                // Prefer unresolved type over inferred string from concat, so that it can later be resolved to literal.
+                $const_type = null;
+            }
+
             if ($const_type) {
                 $existing_constants[$const->name->name] = new \Psalm\Storage\ClassConstantStorage(
                     $const_type,
-                    ClassLikeAnalyzer::VISIBILITY_PUBLIC,
+                    $stmt->isProtected()
+                        ? ClassLikeAnalyzer::VISIBILITY_PROTECTED
+                        : ($stmt->isPrivate()
+                            ? ClassLikeAnalyzer::VISIBILITY_PRIVATE
+                            : ClassLikeAnalyzer::VISIBILITY_PUBLIC),
```

Comment:
Not quite sure if I'm fixing a mistake or introducing a bug here. I _think_ it fixes a mistake, since it looks like these are existing constants for the current class, but none of the tests are affected either way, so I'm not sure exactly what affect this has.

Reply:
This is unnecessary — these constants are not actually used, but it's a harmless change so it's fine to keep.